include/deal.II/base/array_view.h

```diff
@@ -423,48 +425,180 @@ ArrayView<ElementType>::operator[](const std::size_t i) const
 
 
 /**
- * Create a view to an entire C-style array. This is equivalent to
- * initializing an ArrayView object with a pointer to the first element and
- * the size of the given argument.
+ * Create an ArrayView that takes a pair of iterators as arguments. The type
+ * of the ArrayView is inferred from the value type of the iterator (e.g., the
+ * view created from two const iterators will have a const type).
```

Comment:
Is this something you had intended for the other PR? Not that it matters much, just curious.

Reply:
Actually, this is a bit misleading. I didn't mean to change that much in `ArrayView` but need to reorder the functions so I can use the `make_array_view(ElementType *const, ElementType *const)` and the  `make_array_view(const Iterator, const Iterator)` overload also in the other overloads (without forward declarations).

Reply:
I see. No worries.